### PR TITLE
feat(Ranger): Compatible with the old ACL and the new ACL

### PR DESCRIPTION
**Pull request by WHBANG**
### What problem does this PR solve? 
https://github.com/apache/incubator-pegasus/issues/1054

This patch is compatible with old and new acl.

- Modify some method names and parameter names to make them more accurate.
- Defines the configuration parameters that the new ACL needs to use.
- Two new 'allowed()' methods are provided for meta_server and replica_server.
- Some incompatible methods will be removed, (allowed&pre_check)commented.

### CODE REVIEW

File: src/runtime/security/access_controller.cpp [hunk 1]
```diff
@@ -26,11 +26,22 @@
 namespace dsn {
 namespace security {
 DSN_DEFINE_bool(security, enable_acl, false, "whether enable access controller or not");
-DSN_TAG_VARIABLE(enable_acl, FT_MUTABLE);
```
Comment (acelyc111) [inline src/runtime/security/access_controller.cpp:29]: nit: Remove the blank line.

File: src/runtime/security/access_controller.cpp [hunk 2]
```diff
@@ -26,11 +26,22 @@
...
 
-DSN_DEFINE_string(security, super_users, "", "super user for access controller");
+DSN_DEFINE_bool(security, enable_ranger_acl, false, "whether enable access controller or not");
```
Comment (acelyc111) [inline src/runtime/security/access_controller.cpp:30]:
```suggestion
DSN_DEFINE_bool(security, enable_ranger_acl, false, "whether enable access controller integrate to Apache Ranger or not");
```

File: src/runtime/security/access_controller.cpp [hunk 3]
```diff
@@ -26,11 +26,22 @@
...
 
-access_controller::access_controller() { utils::split_args(FLAGS_super_users, _super_users, ','); }
+DSN_DEFINE_string(security,
+                  super_users,
+                  "",
```
Comment (acelyc111) [inline src/runtime/security/access_controller.cpp:34]: What happen if enable acl but leave super_users as empty?

File: src/runtime/security/access_controller.cpp [hunk 4]
```diff
@@ -40,6 +51,14 @@ bool access_controller::pre_check(const std::string &user_name)
         return true;
     }
     return false;
+    return _super_users.find(user_name) != _super_users.end();
+}
+
+bool access_controller::is_enable_ranger_acl() { return FLAGS_enable_ranger_acl; }
```
Comment (acelyc111) [inline src/runtime/security/access_controller.cpp:57]: Why not use `FLAGS_enable_ranger_acl` directly?

File: src/runtime/security/access_controller.h [hunk 5]
```diff
@@ -31,27 +33,41 @@ class access_controller
...
-    /**
-     * check if the message received is allowd to do something.
-     *   msg - the message received
-     **/
+    // Check if the message received is allowd to access the table.
+    // msg - the message received
+    // app_name - tables involved in ACL
+    virtual bool allowed(message_ex *msg, const std::string &app_name) { return false; }
+
+    // (TODO:wanghao) : this method will be deleted in the next patch.
```
Comment (acelyc111) [inline src/runtime/security/access_controller.h:52]:
```suggestion
    // TODO(wanghao): this method will be deleted in the next patch.
```

File: src/runtime/security/access_controller.h [hunk 6]
```diff
@@ -31,27 +33,41 @@ class access_controller
...
+    // check if the message received is allowd to do something.
+    // msg - the message received
     virtual bool allowed(message_ex *msg) = 0;
 
 protected:
+    // (TODO:wanghao) : this method will be deleted in the next patch.
```
Comment (acelyc111) [inline src/runtime/security/access_controller.h:58]:
```suggestion
    // TODO(wanghao): this method will be deleted in the next patch.
```

File: src/runtime/security/access_controller.cpp (hunk 4 above)
Comment (WHBANG) [reply src/runtime/security/access_controller.cpp:57]: Both `dsn_meta_server` and `dsn_replica_server` will use it, so `FLAGS_enable_ranger_acl` cannot be used directly.

**Review by acelyc111 — APPROVED** [review]

File: src/runtime/security/access_controller.cpp (hunk 4 above)
Comment (WHBANG) [reply src/runtime/security/access_controller.cpp:57]: yes

**Review by empiredan — APPROVED** [review]

**Pull request merged**